caster.go

```diff
@@ -0,0 +1,33 @@
+package qkit
+
+import (
+	"encoding/json"
+)
+
+// Converts any type to a given type. If conversion fails, it returns the zero value of the given type.
+func Cast[T any](val any) T {
+	if val, ok := val.(T); ok {
+		return val
+	}
+	var zero T
+	return zero
+}
+
+// Converts any type to a given type. This partially fills the target in case the 2 types are not directly compatible.
+func CastPartial[T any](val any) T {
+	return FromBytes[T](Ok(json.Marshal(val)))
+}
+
+// Converts any type to a map[string]interface{}.
+func ToMap(s any) map[string]interface{} {
+	m := make(map[string]interface{})
+	json.Unmarshal(Ok(json.Marshal(s)), &m)
+	return m
+}
+
+// Converts a byte array to a given type.
+func FromBytes[T any](bytes []byte) T {
```

Comment:
also this could be `FromJSON`

Reply:
Done. I also added back the previous `ToBytes` as `ToJSON`. Please do let me know if there's any more to be done. Further I noticed that this repo doesn't have a workflow for unit tests. I could add it if a future PR if you do need one. Since the repo's public it will not consume and build minutes.